lib/tasks/db.rake

```diff
@@ -61,6 +61,77 @@ namespace :db do
     end
   end
 
+  namespace :search do
+    require 'optparse'
+
+    desc 'Creates search indexes for lucend style queries.'
+    task :create, [:db, :silent, :analyzer] => :environment do |_t, args|
+      options = {}
+      opts = OptionParser.new
+      opts.on("-d", "--db ARG", String) { |db| options[:db] = db }
+      opts.on("-s", "--silent", TrueClass) { options[:silent] = true }
+      opts.on("-a", "--analyzer ARG", String) { |analyzer| options[:analyzer] = analyzer }
+      args = opts.order!(ARGV) {}
+      opts.parse!(args)
+
+      analyzer = options[:analyzer].presence || 'standard'
+      db = options[:db].present? ? "/#{options[:db]}/" : ''
+      silent = options[:silent].present?
+
+      design_dir = Rails.root.join 'db', 'searches'
+      files = Dir.glob File.join design_dir, '**', '*.js'
+
+      puts "\n\e[44m== Updating search indexes ==\e[0m\n\n" unless silent
+
+      docs = files.map do |filename, acc|
+        name  = File.basename(filename).sub(/.js/i, '')
+        source = File.read filename
+        design_doc_name = "_design/#{name}"
+
+        doc = begin
+                Dolly::Document.connection.request(:get, "#{db}#{design_doc_name}")
+              rescue Dolly::ResourceNotFound
+                {}
+              end
+
+        data = {
+          _id: design_doc_name,
+          indexes: {
+            name => {
+              index: source,
+              analyzer: analyzer
+            }
+          }
+        }
+
+        if data[:indexes].to_json == doc[:indexes].to_json
+          puts "\e[32mSearch index #{name} is up to date.\e[0m" unless silent
+          {}
+        else
+          puts "\e[36mSearch Index #{name} will be updated.\e[0m" unless silent
+          doc.merge!(data)
+        end
+      end
+
+      res = Dolly::Document.connection.request :post, "#{db}_bulk_docs", docs: docs
```

Comment:
I think you missed a `/`. If a db name is passed the url should be `#{db}/_bulk_docs`? Or am I missing something?

Reply:
Nop, it is like this so if no db is setup dolly will peak the default db and no initial `/` is required

Reply:
Nevermind, my bad.

Reply:
I didnt see you add the / in the db variable.